src/cmap/auth/mongodb_oidc/cache.ts

```diff
@@ -1,7 +1,48 @@
+/* 5 minutes in milliseconds */
+const EXPIRATION_BUFFER_MS = 300000;
+
+/**
+ * An entry in a cache that can expire in a certain amount of time.
+ */
+export abstract class ExpiringCacheEntry {
```

Comment:
Creating this class as we now have 2 caches that have expiring entries of the same expiration time. (Azure, Callbacks)